compiler/resolution/preFold.cpp

```diff
@@ -115,7 +115,10 @@ Expr* preFold(CallExpr* call) {
       call->replace(retval);
 
     } else {
-      if (isLcnSymbol(symExpr->symbol()) == true) {
+      if (symExpr->symbol()->hasFlag(FLAG_TYPE_VARIABLE) &&
+          symExpr->getValType()->symbol->hasFlag(FLAG_TUPLE) == false) {
```

Comment:
Checking for tuple here seems again to be a workaround, but I might be wrong about that. I'm not sure I understand how tuple type construction works in this PR. If there is future work in this area, could you make a note about it in the PR message?

Reply:
Tuple type construction still goes through 'createTupleSignature', I'll add a comment and update the PR message.